plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js

```diff
@@ -94,111 +95,141 @@ const test = baseTest.extend( {
+			await page.getByRole( 'button', { name: 'Add new' } ).click();
 
-			// Add another attribute
-			await page.getByLabel( 'Add another attribute' ).click();
 			await page
-				.getByPlaceholder( 'Search or create attribute' )
-				.nth( 1 )
-				.click();
+				.getByRole( 'heading', { name: 'Add variation options' } )
+				.isVisible();
 
-			await waitForAttributeList( page );
+			await page.waitForLoadState( 'domcontentloaded' );
+		} );
 
-			await page
-				.getByPlaceholder( 'Search or create attribute' )
-				.nth( 1 )
-				.fill( newAttributes[ 1 ].name );
-			await page
-				.getByText( `Create "${ newAttributes[ 1 ].name }"` )
-				.click();
+		await test.step( 'create local attributes with terms', async () => {
+			/*
+			 * attributeRowsLocator are the rows that contains
+			 * the Attribute combobox and the Term FormTokenField.
+			 */
+			const attributeRowsLocator = page.locator(
+				'.woocommerce-new-attribute-modal__table-row'
+			);
 
-			await page.getByPlaceholder( 'Search or create value' ).click();
-			await page
-				.getByPlaceholder( 'Search or create value' )
-				.fill( newAttributes[ 1 ].value );
-			await page
-				.getByText( `Create "${ newAttributes[ 1 ].value }"` )
-				.click();
+			/*
+			 * First, check the app loads the attributes,
+			 * based on the Spinner visibility.
+			 */
+			const spinnerLocator = attributeRowsLocator.locator(
+				'.components-spinner'
+			);
+			await spinnerLocator.waitFor( {
+				state: 'visible',
+			} );
+			await spinnerLocator.waitFor( { state: 'hidden' } );
+
+			for ( const term of newAttributes ) {
+				const attributeRowLocator = attributeRowsLocator.last();
+
+				const attributeComboboxLocator = attributeRowLocator
+					.locator(
+						'input[aria-describedby^="components-form-token-suggestions-howto-combobox-control"]'
+					)
+					.last();
+
+				// Create new (local) product attribute.
+				await attributeComboboxLocator.fill( term.name );
+				await page.locator( `text=Create "${ term.name }"` ).click();
+
+				const FormTokenFieldLocator = attributeRowLocator.locator(
+					'td.woocommerce-new-attribute-modal__table-attribute-value-column'
+				);
+
+				// Term FormTokenField input locator
+				const FormTokenFieldInputLocator =
+					FormTokenFieldLocator.locator(
+						'input[id^="components-form-token-input-"]'
+					);
+
+				// Add terms to the attribute.
+				for ( const attributeTerm of term.terms ) {
+					await FormTokenFieldInputLocator.fill( attributeTerm );
+					await FormTokenFieldInputLocator.press( 'Enter' );
+				}
 
-			// Add attributes
+				await page.getByLabel( 'Add another attribute' ).click();
+			}
+
+			// Add the product attributes
 			await page
 				.getByRole( 'button', { name: 'Add attributes' } )
 				.click();
 		} );
 
 		await test.step( 'verify attributes in product editor', async () => {
-			// Verify new attributes in product editor
+			// Select the main locator by data-template-block-id="product-attributes-section"
```

Comment:
Putting the selector in the comment is unnecessary and likely to get out of date at some point